player/sub.c

```diff
@@ -100,33 +100,36 @@ static bool update_subtitle(struct MPContext *mpctx, double video_pts,
         sub_preload(dec_sub);
     }
 
-    if (!sub_read_packets(dec_sub, video_pts, mpctx->paused))
-        return false;
-
-    // Handle displaying subtitles on terminal; never done for secondary subs
-    if (mpctx->current_track[0][STREAM_SUB] == track && !mpctx->video_out) {
-        char *text = sub_get_text(dec_sub, video_pts, SD_TEXT_TYPE_PLAIN);
-        term_osd_set_subs(mpctx, text);
-        talloc_free(text);
-    }
+    bool packets_read = false;
+    bool sub_updated = false;
+    sub_read_packets(dec_sub, video_pts, mpctx->paused, &packets_read, &sub_updated);
+
+    // Check if we need to update subtitles for these special cases. Always
+    // update on discontinuities like seeking.
+    if (sub_updated || fabs(mpctx->last_seek_pts - video_pts) < 0.01) {
```

Comment:
> 0.01
Is this in seconds? Looks like quite large margin.

Reply:
Yeah it's in seconds. I wanted to err on the slightly larger side since there's seeking always requires an update. I wasn't really sure how big the difference between `mpctx->last_seek_pts` and `video_pts` can get. Of course if there's a better way to detect a seek from here, that would work too.

Reply:
Ended up keeping this in the `mpctx` now.